test/rpc_test.py

```diff
@@ -1319,6 +1319,26 @@ def test_debug_info(self):
         expected.update(autograd_info)
         self.assertEqual(expected, info)
 
+    @requires_process_group_agent("Testing exceptions thrown by ProcessGroupAgent::enqueueSend.")
+    def test_sender_exceptions(self):
```

Comment:
Can we have a common test which is run for all backends and ensures we receive exceptions when other nodes exit? For matching in assertRaisesRegex, we can match the appropriate message based on the backend type?

Reply:
Added, now the message is based on a check for the particular type of agent.